packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -804,7 +804,8 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
         return;
       }
       if (this.props.multiSelect) {
-        option.selected = !option.selected;
+        //Setting the initial state of option.selected in Multi-select combobox by checking the selectedIndices array and overriding the undefined issue
+        option.selected = option.selected != null ? !option.selected : !(selectedIndices.indexOf(index) >= 0);
```

Comment:
Are you sure you want loose equality here? `option.selected != null` will also match `undefined`.

Nit: please reduce negative logic for readability, like using `===` for check and `selectedIndices.indexOf(index) < 0`.